### ISSUE TITLE
APT package repo does not contain `loki` and `promtail` packages

### ISSUE THREAD

**Issue by MAFLO321** [grafana/loki#7705]
**Describe the bug**
Official Debian package repo of Grafana does not contain `loki` and `promtail` packages.

**To Reproduce**
1. add Debian repo as [described](https://grafana.com/docs/grafana/latest/setup-grafana/installation/debian/)
2. try to install any of the mentioned packages with `apt`
3. the installation fails because
   1. on amd64 platform the [Packages](https://apt.grafana.com/dists/stable/main/binary-amd64/Packages) file contains the packages, but the referenced [package](https://apt.grafana.com/pool/main/l/loki/loki_2.7.0_amd64.deb) [files](https://apt.grafana.com/pool/main/p/promtail/promtail_2.7.0_amd64.deb) do not exist on the server. 
   2. on arm64 platform the [Packages](https://apt.grafana.com/dists/stable/main/binary-arm64/Packages) file contains the packages, but the referenced [package](https://apt.grafana.com/pool/main/l/loki/loki_2.7.0_arm64.deb) [files](https://apt.grafana.com/pool/main/p/promtail/promtail_2.7.0_arm64.deb) do not exist on the server. 
   2. on armhf platform the [Packages](https://apt.grafana.com/dists/stable/main/binary-armhf/Packages) file does not contain any of the loki packages.

**Expected behavior**
The loki packages should be installable by apt on all three platforms.

**Environment:**
 - Debian 11

As a side note, I'm not sure if the same problem also exists for the [rpm repo](https://grafana.com/docs/grafana/latest/setup-grafana/installation/rpm/). If someone on the CD team is addressing this issue, you may want to check it as well.

refs: #52, #6456

**Comment by jeschkies** [comment]
We've recently create a Grafana repository https://apt.grafana.com/ and https://rpm.grafana.com/. Would that work for you?

**Comment by MAFLO321** [comment]
yes, but as described the `armhf`-architecture on https://apt.grafana.com/ is lacking of the `loki` and `promtail` packages.
This can be seen in https://apt.grafana.com/dists/stable/main/binary-armhf/Packages, because it does not contain entries for `loki` and `promtail` packages.

But on github `armhf` compatible packages are released ([loki_2.7.4_arm.deb](https://github.com/grafana/loki/releases/download/v2.7.4/loki_2.7.4_arm.deb), [promtail_2.7.4_arm.deb](https://github.com/grafana/loki/releases/download/v2.7.4/promtail_2.7.4_arm.deb)), with an big but.. The architecture of the .dep package shows `arm` and needs to be changed to `armhf` manually. Then the packages are working on a Raspberry Pi 2 (which has is `armhf` architecture).

**Comment by fnwbr** [comment]
What is the pull request waiting for? Can I help with this?
A year later and the official installation instructions for Loki (https://grafana.com/docs/loki/latest/setup/install/local/#install-using-apt-or-rpm-package-manager) are still not working on `arm` systems, when trying to install via `apt`.

**Comment by gpinkos** [comment]
As @fnwbr mentioned, this is still not working.  Can someone at Grafana explain the reason for such a glaring omission?

**Comment by alexitheodore** [comment]
I have the same issue! I am surprised that more people aren't saying something and that this has gone unaddressed.

**Comment by lucaci32u4** [comment]
I am having the same issue. I am trying to install promtail on RPI4 (32-bit OS) to collect logs from a 3D Printer.

**Comment by Joypaul11** [comment]
Same issue

**Comment by JStickler** [comment]
Promtail is end of life (EOL) as of  March 2, 2026.  No future support or updates will be provided. All future feature development will occur in [Grafana Alloy](https://github.com/grafana/alloy).
